### ISSUE TITLE
docker 1.11.0 crashes after docker-compose kill & rm & up

### ISSUE THREAD

**Issue by schmunk42** [docker/docker#22194]
**Output of `docker version`:**

```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 19:36:04 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 19:36:04 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 38
 Running: 10
 Paused: 0
 Stopped: 28
Images: 257
Server Version: 1.11.0
Storage Driver: aufs
 Root Dir: /mnt/sda1/var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 616
 Dirperm1 Supported: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge null host
Kernel Version: 4.1.19-boot2docker
Operating System: Boot2Docker 1.11.0 (TCL 7.0); HEAD : 32ee7e9 - Wed Apr 13 20:06:49 UTC 2016
OSType: linux
Architecture: x86_64
CPUs: 6
Total Memory: 2.936 GiB
Name: runner-host-9
ID: WPZA:DKEF:U5A5:BRGK:6QY3:FU22:NDZT:X7MM:N7DU:T45M:5ON2:MWEM
Docker Root Dir: /mnt/sda1/var/lib/docker
Debug mode (client): false
Debug mode (server): true
 File Descriptors: 89
 Goroutines: 197
 System Time: 2016-04-20T12:58:23.329654573Z
 EventsListeners: 0
Registry: https://index.docker.io/v1/
Labels:
 provider=virtualbox
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Virtualbox

**Steps to reproduce the issue:**
1. upgrade from 1.10.2 to 1.11.0
2. docker-compose kill && docker-compose rm -fv
3. docker-compose up -d
4. docker-compose run ....

**Describe the results you received:**

Docker daemon hangs. Manual restart takes up to 1 minute.

```
If it's at a non-standard location, specify the URL with the DOCKER_HOST environment variable.
make: *** [run-tests] Error 1
```

See gist, [errors from ci and docker.log](https://gist.github.com/schmunk42/a80e5fdd3d44b50e0e7dc05ada3f5bf3)

**Describe the results you expected:**

No errors or hanging daemon :)

**Additional information you deem important (e.g. issue happens only occasionally):**

It was produceable on a builder in our CI (running with dumb-init). Not always on the exact same task. But almost always in this project which was creating, killing and removing a lot of containers.

**Comment by cpuguy83** [comment]
Thanks. Looks like the daemon logs are missing some of that stack trace. Any way you can grab the full trace?

**Comment by schmunk42** [comment]
@cpuguy83 I've updated the gist.

**Comment by cpuguy83** [comment]
Perfect, thanks!

**Comment by cpuguy83** [comment]
ping @aboch Looks like a resolution panic.

**Comment by aboch** [comment]
Looks like same as https://github.com/docker/libnetwork/issues/1117
@sanimej has the fix for this in https://github.com/docker/libnetwork/pull/1125.

**Comment by adamlc** [comment]
I'm getting the same issue on Debian Jessie, seems to randomly break our server but its the same error on the logs!

**Comment by thaJeztah** [comment]
This will be fixed in 1.11.1 through https://github.com/docker/docker/pull/22261 (which was just merged)
